dataikuapi/dss/ml.py

```diff
@@ -393,40 +392,81 @@ def __repr__(self):
         res += self._key_repr("distributed")
         if self._raw_settings.get("distributed", False):
             res += self._key_repr("nContainers")
-        res += ")"
+        res += ")</pre>"
+        res += "<details><pre>{}</pre></details>".format(self.__repr__())
         return res
-    
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(settings={})".format(json.dumps(self._raw_settings, indent=4))
+
     __str__ = __repr__
 
-    def set_strategy(self, strategy):
-        assert strategy in {"GRID", "RANDOM", "BAYESIAN"}
-        self._raw_settings["strategy"] = strategy
+    def _set_seed(self, seed):
+        if seed is not None:
+            if not isinstance(seed, int):
+                warnings.warn("HyperparameterSearchSettings ignoring invalid input: seed")
+                warnings.warn("seed must be an integer")
+            else:
+                self._raw_settings["seed"] = seed
+
+    def set_strategy_to_grid_search(self, shuffle=None, seed=None):
```

Comment:
`shuffle` and `seed` should default to actual default (if applicable, maybe null `seed` is valid?), makes it simpler & self-documenting.
[Same with all similar methods, including those for validation mode]